src/main/java/org/embulk/input/zendesk/services/ZendeskChatService.java

```diff
@@ -0,0 +1,280 @@
+package org.embulk.input.zendesk.services;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import org.embulk.config.TaskReport;
+import org.embulk.input.zendesk.RecordImporter;
+import org.embulk.input.zendesk.ZendeskInputPlugin;
+import org.embulk.input.zendesk.clients.ZendeskRestClient;
+
+import org.embulk.input.zendesk.utils.ZendeskConstants;
+import org.embulk.input.zendesk.utils.ZendeskDateUtils;
+import org.embulk.input.zendesk.utils.ZendeskUtils;
+import org.embulk.spi.Exec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class ZendeskChatService implements ZendeskService
+{
+    private static final Logger logger = LoggerFactory.getLogger(ZendeskChatService.class);
+
+    private ZendeskRestClient zendeskRestClient;
+
+    protected ZendeskInputPlugin.PluginTask task;
+
+    private static final int MAXIMUM_TOTAL_RECORDS = 10000;
+
+    private static final int MAXIMUM_TOTAL_PAGES = 250;
+
+    private static final int MAXIMUM_RECORDS_PER_PAGE = 40;
+
+    public ZendeskChatService(final ZendeskInputPlugin.PluginTask task)
+    {
+        this.task = task;
+    }
+
+    @Override
+    public boolean isSupportIncremental()
+    {
+        return true;
+    }
+
+    @Override
+    public TaskReport addRecordToImporter(final int taskIndex, final RecordImporter recordImporter)
+    {
+        final TaskReport taskReport = Exec.newTaskReport();
+        String startTime = getStartTime();
+        String endTime = getEndTime();
+
+        if (Exec.isPreview()) {
+            fetchData(startTime, endTime, 1, recordImporter);
+            return taskReport;
+        }
+
+        while (true) {
+            String searchURI = buildSearchRequest(startTime, endTime, 1);
+
+            String response = getZendeskRestClient().doGet(searchURI, task, Exec.isPreview());
+
+            JsonNode json = ZendeskUtils.parseJsonObject(response);
+
+            if (!ZendeskUtils.isNull(json)) {
+                int totalRecords = json.get("count").asInt();
+                int totalPages = totalRecords > MAXIMUM_TOTAL_RECORDS
+                    ? MAXIMUM_TOTAL_PAGES
+                    : slicePages(totalRecords);
+
+                logger.info(String.format("Fetching from '%s' to '%s' with '%d' pages and '%d' records",
+                    startTime, endTime, totalPages,  totalRecords));
+
+                if (totalPages > 1) {
```

Comment:
I'm not really sure why do we have branching here? Tried with:
```
IntStream.range(1, 2).parallel().forEach(System.out::println);
// totalPages = 1
// 2 = totalPages + 1
```
and the result is 1, 2nd parameter is exclusive.

Reply:
Yah my mistake during refactoring

Reply:
fixed

Reply:
how can you refactor the new file 😂 . Anyway, thanks for fixing.